horace_core/sqw/@sqw/join.m

```diff
@@ -17,162 +17,134 @@
-wout.detpar=detpar0;
+wout.main_header = main_header;
+wout.experiment_info = Experiment(header); % This should be a cell array of the individual headers
+wout.detpar = detpar0;
 
 first_included_index = find(run_contributes,1,'first'); % find first included file to ensure non-empty data below
-wout.data=data{run_contributes(first_included_index)};
-sz=size(wout.data.npix); % size of contributing signal, variance, and npix arrays
-wout.data.s   =zeros(sz);
-wout.data.e   =zeros(sz);
-wout.data.npix=zeros(sz);
-% build a new PixelData object from the contributing files data
-pix_ex = cellfun(@(x) x, pix(run_contributes), 'UniformOutput', false);
-[wout, ldr] = wout.get_new_handle();
-% TODO:  Re #1032  This is incorrect. Despite split is reversed, the pixels are not 
-% combined in correct order.
-wout.pix = wout.pix.cat(pix_ex{:}, ldr);
-for i=1:nfiles
+wout.data = data{run_contributes(first_included_index)};
+sz = size(wout.data.npix); % size of contributing signal, variance, and npix arrays
+wout.data.s = zeros(sz);
+wout.data.e = zeros(sz);
+wout.data.npix = zeros(sz);
+
+for i = 1:nfiles
     if run_contributes(i)
-        wout.data.s   = wout.data.s   + (data{i}.s).*(data{i}.npix);
-        wout.data.e   = wout.data.e   + (data{i}.e).*(data{i}.npix).^2;
-        wout.data.npix= wout.data.npix+ data{i}.npix;
+        wout.data.s = wout.data.s + (data{i}.s .* data{i}.npix);
+        wout.data.e = wout.data.e + (data{i}.e .* (data{i}.npix .^ 2));
+        wout.data.npix = wout.data.npix + data{i}.npix;
     end
 end
+
 wout.data.s = wout.data.s ./ wout.data.npix;
-wout.data.e = wout.data.e ./ (wout.data.npix).^2;
-wout.data.s(~wout.data.npix)=0;
-wout.data.e(~wout.data.npix)=0;
-
-% It seems that sorting of the pix field is necessary, however I'm not
-% aware of an easy way of accomplishing this.
-% Re-performing a cut with
-%   wout=cut_sqw(wout,proj,[],[],[],[]);
-% does the trick, but that is projection-specific and it might not be right
-% to build up a proj struct from the information present in a sqw.
-%
-% An alternative is to use the projection-axis cut form.
-%TODO: Re #1032 This have to be done on 0-dimension d0d object as keys. This is not currently the case
-%   wout=cut_sqw(wout,[]...)
-% wrapped with commands to temporarily silence Horace.
-hc_log_level=get(hor_config,'log_level');
-set(hor_config,'log_level',-1);
-
-cut_args = repmat({[]},size(wout.data.p));
-wout=cut_sqw(wout,cut_args{:});
-set(hor_config,'log_level',hc_log_level);
+wout.data.e = wout.data.e ./ (wout.data.npix .^ 2);
+wout.data.s(~wout.data.npix) = 0;
+wout.data.e(~wout.data.npix) = 0;
+
+% build a new PixelData object from the contributing files data
+% pix_ex = cellfun(@(x) x, pix(run_contributes), 'UniformOutput', false);
+
+ind = zeros(sz);
+num_pix = sum(wout.data.npix, 'all');
+wout.pix = PixelDataMemory(num_pix);
+
+for i = 1:nfiles
+    if run_contributes(i)
```

Comment:
very inefficient implementation. This one is slow up to uslesness due to the way Matlab treats loops and array indexing. Much more reasonable would be swapping order of loops, i.e. collecting all contributed pixels from different files and bunching them all together to put them into target file. 

Would assume this is a quick fix to solve the problem, but more efficient implementations are already in Horace -- `write_nsqw_to_sqw` works (Maltab and C++ implementations) do the same for files and expanding them to memory based implementation is a trivial excersize. 

(May also be beneficial to look on Matlab `write_nsqw_to_sqw` from point of view of using generic interface to PixelData -- then the combine code will be applicable in both places)

Inabilty to use existing highly optimized code in favour of custom half-baked solutions is depressing